codegenerator/cli/templates/static/codegen/src/Benchmark.res

```diff
@@ -115,25 +165,98 @@ module Data = {
     self.millisAccum->MillisAccum.increment(label, amount)
   }
 
+  module LiveMetrics = {
+    module ThrottlerMapping = {
+      module LabelToThrottlerMapping = {
+        type t = dict<Throttler.t>
+        let make = (): t => Js.Dict.empty()
+        let get = (self: t, ~group: string, ~label: string) =>
+          switch self->Utils.Dict.dangerouslyGetNonOption(label) {
+          | Some(throttler) => throttler
+          | None =>
+            let throttler = Throttler.make(
+              ~intervalMillis=Env.ThrottleWrites.liveMetricsBenchmarkIntervalMillis,
+              ~logger=Logging.createChild(
+                ~params={
+                  "context": "Throttler for live metrics benchmarking",
+                  "group": group,
+                  "label": label,
+                },
+              ),
+            )
+            self->Js.Dict.set(label, throttler)
+            throttler
+          }
+      }
+      type t = dict<LabelToThrottlerMapping.t>
+
+      let make = (): t => Js.Dict.empty()
+
+      let get = (self: t, ~group: string, ~label: string): Throttler.t => {
+        let labelToThrottlerMapping = switch self->Utils.Dict.dangerouslyGetNonOption(group) {
+        | Some(labelToThrottlerMapping) => labelToThrottlerMapping
+        | None =>
+          let labelToThrottlerMapping = LabelToThrottlerMapping.make()
+          self->Js.Dict.set(group, labelToThrottlerMapping)
+          labelToThrottlerMapping
+        }
+        labelToThrottlerMapping->LabelToThrottlerMapping.get(~group, ~label)
+      }
+    }
+
+    let saveLiveMetrics = if (
+      Env.Benchmark.saveDataStrategy->Env.Benchmark.SaveDataStrategy.shouldSavePrometheus
+    ) {
+      let throttlerMapping = ThrottlerMapping.make()
+
+      (dataSet: SummaryData.DataSet.t, ~group, ~label) => {
+        let throttler = throttlerMapping->ThrottlerMapping.get(~group, ~label)
+        throttler->Throttler.schedule(() => {
+          let {n, mean, stdDev, min, max, sum} = dataSet->Stats.makeFromDataSet
+          Prometheus.setBenchmarkSummaryData(
+            ~group,
+            ~label,
+            ~n,
+            ~mean,
+            ~stdDev,
+            ~min,
+            ~max,
+            ~sum,
+          )->Promise.resolve
+        })
+      }
+    } else {
+      (_dataSet, ~group as _, ~label as _) => ()
+    }
```

Comment:
If `ENVIO_SAVE_BENCHMARK_DATA_STRATEGY="prometheus"` this will return true. I black boxed the strategy type since we may want to introduce other things like saving to postgres orboth prometheus and another type etc.